lib/sgpres.gi

```diff
@@ -2766,38 +2766,1163 @@ end);
+    fi;
+  end;
+
+  ct:=DATA.ct;
+  offset:=DATA.offset;
+  p:=DATA.p;
+  l:=0;
+  q:=[];
+  Merge(a,b);i:=1;
+  while i<=l do
+    c:=q[i];
+    i:=i+1;
+    #RemoveSet(DATA.omega,c);
+    for x in DATA.A do
+      if ct[x+offset][c]<>0 then
+	d:=ct[x+offset][c];
+	ct[-x+offset][d]:=0;
+	mu:=Rep(c);
+	nu:=Rep(d);
+	if ct[x+offset][mu]<>0 then
+	  Merge(nu,ct[x+offset][mu]);
+	elif ct[-x+offset][nu]<>0 then
+	  Merge(mu,ct[-x+offset][nu]);
+	else
+	  ct[x+offset][mu]:=nu;
+	  ct[-x+offset][nu]:=mu;
+	  Add(DATA.deductions,[mu,x]);
+	fi;
+      fi;
+    od;
+  od;
+  #Error("coinc\n");
+end;
+
+NEWTC_ModifiedCoincidence:=function(DATA,a,b,w)
+local MRep,MMerge,ct,offset,l,q,i,c,x,d,p,pp,mu,nu,aug,v,Sekundant;
+
+  Sekundant:=function(w)
+    if Length(w)<=1 then
+      return w;
+    fi;
+    DATA.secount:=DATA.secount+1;
+    DATA.secondary[DATA.secount]:=w;
+    return [DATA.secount];
+  end;
+
+  MRep:=function(kappa)
+  local lambda,rho,mu,s;
+    lambda:=kappa;
+    rho:=p[lambda];
+    if rho=lambda then
+      return lambda;
+    fi;
+
+    s:=[]; # new array to trace back compression path
+    while rho<>lambda do
+      s[rho]:=lambda;
+      lambda:=rho;rho:=p[lambda];
+    od;
+    rho:=s[lambda];
+    while rho<>kappa do
+      mu:=rho;
+      rho:=s[mu];
+      p[rho]:=lambda;
+      pp[rho]:=Sekundant(WordProductLetterRep(pp[rho],pp[mu]));
+    od;
+    return lambda;
+  end;
+
+  MMerge:=function(k,a,w)
+  local phi,psi,mu,nu;
+    phi:=MRep(k);
+    psi:=MRep(a);
+    if phi>psi then
+      p[phi]:=psi;
+      pp[phi]:=Sekundant(WordProductLetterRep(-Reversed(pp[k]),w,pp[a]));
+      l:=l+1;
+      q[l]:=phi;
+      DATA.dead:=DATA.dead+1;
+    elif psi>phi then
+      p[psi]:=phi;
+      pp[psi]:=Sekundant(WordProductLetterRep(-Reversed(pp[a]),-Reversed(w),pp[k]));
+      l:=l+1;
+      q[l]:=psi;
+      DATA.dead:=DATA.dead+1;
+    fi;
+  end;
+
+  ct:=DATA.ct;
+  aug:=DATA.aug;
+  offset:=DATA.offset;
+  p:=DATA.p;
+  pp:=DATA.pp;
+  l:=0;
+  q:=[];
+  MMerge(a,b,w);i:=1;
+  while i<=l do
+    c:=q[i];
+    i:=i+1;
+    for x in DATA.A do
+      if ct[x+offset][c]<>0 then
+	d:=ct[x+offset][c];
+	ct[-x+offset][d]:=0;
+	mu:=MRep(c);
+	nu:=MRep(d);
+	if ct[x+offset][mu]<>0 then
+	  v:=WordProductLetterRep(-Reversed(pp[d]),-Reversed(aug[x+offset][c]),
+		pp[c],aug[x+offset][mu]);
+	  MMerge(nu,ct[x+offset][mu],v);
+	elif ct[-x+offset][nu]<>0 then
+	  v:=WordProductLetterRep(-Reversed(pp[c]),aug[x+offset][c],
+		pp[d],aug[-x+offset][nu]);
+	  MMerge(mu,ct[-x+offset][nu],v);
+	else
+	  ct[x+offset][mu]:=nu;
+	  ct[-x+offset][nu]:=mu;
+	  v:=WordProductLetterRep(-Reversed(pp[c]),aug[x+offset][c],pp[d]);
+	  aug[x+offset][mu]:=v;
+	  aug[-x+offset][nu]:=-Reversed(v);
+	  Add(DATA.deductions,[mu,x]);
+	fi;
+      fi;
+    od;
+  od;
+  # pp is not needed any longer
+  for i in q do
+    Unbind(pp[i]);
+  od;
+end;
+
+# superseded by kernel function, left here for debugging purposes.
+NEWTC_ObsoleteFailScan:=function(c,offset,alpha,w)
```

Comment:
At least I *guess* that `TC_QUICK_SCAN` is meant -- but looking at it close, it turns out `TC_QUICK_SCAN` has a different signature, with an additional `result` argument, which apparently returns some data. So, it's not clear to me at all what this function is about, and how one would use it for debugging purposes...